WordPress/src/main/java/org/wordpress/android/models/Role.java

```diff
@@ -21,21 +27,42 @@
         this.backgroundColor = backgroundColor;
     }
 
-    public static String toString(Context context, Role role) {
+    // Label to be used in UI
+    public static String getLabel(Context context, Role role) {
         return context.getString(role.label);
     }
 
-    // This method will be used to determine the role of the user from network request
-    public static Role fromString(String value) {
+    // Returns the key for that role
+    public static String toKey(Role role) {
+        switch (role) {
+            case SUPER_ADMIN:
+                return KEY_SUPER_ADMIN;
+            case ADMIN:
+                return KEY_ADMIN;
+            case EDITOR:
+                return KEY_EDITOR;
+            case AUTHOR:
+                return KEY_AUTHOR;
+            case CONTRIBUTOR:
+                return KEY_CONTRIBUTOR;
+            default:
```

Comment:
It's a conscious decision to keep the "unsupported" string even if the input is `null`. `toKey(Role)` method is used to convert the role to something we can write to DB, and this is the best way to do it (IMO). Unsupported is not an actual role, if we are fine to handle `null` everywhere `Role` is used, there is no need for it anyway. So all in all, I think this is probably the best version we'll have which solves the issue you brought up in your initial comment. Thanks!